### PR TITLE
Restore logic to set UA suffixes

### PR DESCRIPTION
**Pull request by bdonlan**
The logic to set the UA suffix to include the SDK version was dropped as part of the new builder refactoring. This change restores it.

Also included is a minor fix to a test (this didn't break anything but should be fixed out of general tidiness).

### CODE REVIEW

File: src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java [hunk 1]
```diff
@@ -51,6 +57,75 @@
...
+
+    private final String userAgent_;
+
+    static {
+        Field clientConfiguration = null;
+        try {
+            clientConfiguration = AmazonWebServiceClient.class.getDeclaredField("clientConfiguration");
+            clientConfiguration.setAccessible(true);
+        } catch (Throwable t) {
+            // Incompatible SDK change, we'll leave the Field null. Later we'll just replace the entire UA.
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java]: Please set `clientConfiguration` to `null` in this catch block.

File: src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java [hunk 2]
```diff
@@ -51,6 +57,75 @@
...
+        }
+
+        f_clientConfiguration = clientConfiguration;
+    }
+
+    private static String getOriginalUserAgent(AWSKMS kms) {
+        try {
+            ClientConfiguration clientConfiguration = null;
+            if (kms instanceof AWSKMSClient) {
+                clientConfiguration = (ClientConfiguration)f_clientConfiguration.get(kms);
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java]: What if `f_clientConfiguration == null`?

NitPick: I know we don't do it everywhere, but I much prefer a space after a cast.

File: src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java [hunk 3]
```diff
@@ -51,6 +57,75 @@
...
+
+                if (suffix == null) {
+                    suffix = "";
+                }
+
+                return prefix + suffix;
+            }
+        } catch (Throwable t) {
+            // fall through
+            // note: Throwable catch is required because we are calling invokeExact
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java]: This is no longer accurate. Please fix the comment and the catch-block.

File: src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java [hunk 4]
```diff
@@ -51,6 +57,75 @@
...
+
+    private static String getUserAgent(AWSKMS kms) {
+        String ua;
+        String baseUserAgent = getOriginalUserAgent(kms);
+
+        if (baseUserAgent == null) {
+            // The internal SDK fields we use to extract the base user agent are missing. We'll override the UA entirely.
+            ua = VersionInfo.USER_AGENT;
+        } else if (baseUserAgent.isEmpty()) {
+            // We need to ensure we don't start with an whitespace character as this breaks client-side signing logic.
```
Comment (SalusaSecondus) [inline src/main/java/com/amazonaws/encryptionsdk/kms/KmsMasterKey.java]: Where do we ensure this? Not here, so the comment here isn't the right place.

**Comment by bdonlan** [comment]
After discussion with the Java SDK folks, I found a simpler way to update the UA. The appendUserAgent API is documented as internal but they say we should be fine to use it.

**Pull request merged**